libraries/scripting/common/src/kotlin/script/experimental/api/linkedPushStack.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package kotlin.script.experimental.api
+
+interface ILinkedPushStack<T> {
```

Comment:
Considering `I...` naming, what about names in this file `compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/repl/ReplState.kt` ?

Reply:
Renamed invoke operator to `get` method

Reply:
Reasons of that it's not a list is that we have more complicated, nonlinear structure: every eval snippet refers both previous eval snippet and corresponding compile snippet (which also has the same structure of list). List which we need is a kind of functional list which consists of head (first n-1 elements) and tail (the last element), and head should have same type, recursively. Plain links are bad because implementation becomes repetitive and more wordy. Let's discuss it again then.

Reply:
> Considering `I...` naming, what about names in this file `compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/repl/ReplState.kt` ?

This is an outlier that came partially with external PR. We're not generally using this convention in the codebase.